task/deprecated-image-check/0.4/deprecated-image-check.yaml

```diff
@@ -53,20 +55,39 @@ spec:
         trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
 
         IMAGES_TO_BE_PROCESSED_PATH="/tmp/images_to_be_processed.txt"
-        SBOM_FILE_PATH="/tmp/sbom.json"
+        touch /tmp/images_to_be_processed.txt
 
         success_counter=0
         failure_counter=0
         error_counter=0
         warnings_counter=0
 
-        # Get base images from SBOM
-        cosign download sbom "${IMAGE_URL}@${IMAGE_DIGEST}" > ${SBOM_FILE_PATH}
+        images_processed_template='{"image": {"pullspec": "'"$IMAGE_URL"'", "digests": [%s]}}'
+        digests_processed=()
 
-        cat ${SBOM_FILE_PATH} | jq -r '.formulation? // empty | .[] | .components? // empty | .[] | select(any((.properties // empty)[]; .name | test("^konflux:container:is_(base|builder)_image"))) | .name' > ${IMAGES_TO_BE_PROCESSED_PATH}
-        echo "Detected base images from SBOM:"
-        cat "${IMAGES_TO_BE_PROCESSED_PATH}"
-        echo ""
+        imagewithouttag=$(echo -n $IMAGE_URL | sed "s/\(.*\):.*/\1/")
+        # strip new-line escape symbol from parameter and save it to variable
+        imageanddigest=$(echo $imagewithouttag@$IMAGE_DIGEST)
+
+        # Get the arch and image manifests by inspecting the image. This is mainly for identifying image indexes
+        image_manifests=$(get_image_manifests -i ${imageanddigest})
+        if [ -n "$image_manifests" ]; then
+          while read -r arch arch_sha; do
+            SBOM_FILE_PATH=$(echo "/tmp/sbom-$arch.json")
+            arch_imageanddigest=$(echo $imagewithouttag@$arch_sha)
+
+            # Get base images from SBOM
+            cosign download sbom $arch_imageanddigest > ${SBOM_FILE_PATH}
```

Comment:
No error handling is needed after this one?

Reply:
added logic to skip the loop in-case the task is unable to download the sbom for an arch. This would mean the digest wouldn't be added to `IMAGES_PROCESSED` result and EC would fail the policy check.